src/main/java/billgates/view/gui/RegisterDialog.java

```diff
@@ -0,0 +1,111 @@
+package billgates.view.gui;
+
+import javax.swing.*;
+import java.awt.*;
+import java.util.Arrays;
+
+public class RegisterDialog extends JDialog {
+    JTextField nameInput = new JTextField(17);
+    JTextField nickInput = new JTextField(17);
+    JPasswordField passwordInput = new JPasswordField(17);
+    JPasswordField confirmPasswordInput = new JPasswordField(15);
+    JTextField emailInput = new JTextField(17);
+
+    JButton createButton = new JButton("Create Account");
+    JButton cancelButton = new JButton("Cancel");
+
+    private boolean accepted = false;
+
+    public RegisterDialog(Window owner){
+        super(owner);
+
+        // Set the font
+        this.setFont(new Font("Modern No. 20", Font.BOLD, 14));
+
+        // Set the root panel and its layout
+        JPanel root = new JPanel();
+        this.setContentPane(root);
+        root.setLayout(new BorderLayout());
+
+        // Components
+        if (true){
+            JPanel panel = new JPanel();
+            root.add(panel, BorderLayout.CENTER);
+
+            JLabel nameLabel = new JLabel("<HTML><font color=black>Name</font><font color=red>*</font>" +
+                    "<font color=black>:</font></html>");
+            JLabel nickLabel = new JLabel("Nickname:");
+            JLabel passwordLabel = new JLabel("<HTML><font color=black>Password</font><font color=red>*</font>" +
+                    "<font color=black>:</font></html>");
+            JLabel confirmPasswordLabel = new JLabel("<HTML><font color=black>Confirm password</font>" +
+                    "<font color=red>*</font><font color=black>:</font></html>");
+            JLabel emailLabel = new JLabel("Email:");
+            panel.add(nameLabel);
+            panel.add(nameInput);
+            panel.add(nickLabel);
+            panel.add(nickInput);
+            panel.add(passwordLabel);
+            panel.add(passwordInput);
+            panel.add(confirmPasswordLabel);
+            panel.add(confirmPasswordInput);
+            panel.add(emailLabel);
+            panel.add(emailInput);
+
+            passwordInput.setEchoChar('*');
+            confirmPasswordInput.setEchoChar('*');
+        }
+
+        if (true){
```

Comment:
This is just because I don't want to come up with a new name for the new JPanel. There are two JPanel with the same name "panel". So, if we put this into the if (true){} block, no errors will be raised.